mne/decoding/time_gen.py

```diff
@@ -914,3 +953,358 @@ def time_generalization(epochs_list, clf=None, cv=5, scoring="roc_auc",
                       for train, test in cv)
     scores = np.mean(scores, axis=0)
     return scores
+
+
+class TimeDecoding(_Decoder):
+    """
+    Cross-validate a series of estimators across every time point
+    to decode the event_id of each epoch.
+
+    Parameters
+    ----------
+    cv : int | object
+        If an integer is passed, it is the number of folds.
+        Specific cross-validation objects can be passed, see
+        sklearn.cross_validation module for the list of possible objects.
+        Defaults to 5.
+
+    clf : object | None
+        An estimator compliant with the scikit-learn API (fit & predict).
+        If None the classifier will be a standard pipeline including
+        StandardScaler and a linear SVM with default parameters.
+
+    predict_type : {'predict', 'predict_proba', 'decision_function'}
+        Indicates the type of prediction:
+            'predict' : generates a categorical estimate of each trial.
+
+            'predict_proba' : generates a probabilistic estimate of each trial.
+
+            'decision_function' : generates a continuous non-probabilistic
+                estimate of each trial.
+        Default: 'predict'
+
+    predict_mode : {'cross-validation', 'mean-prediction'}
+        Indicates how predictions are achieved with regards to the cross-
+        validation procedure:
+
+            ``cross-validation`` : estimates a single prediction per sample
+                based on the unique independent classifier fitted in the
+                cross-validation.
+            ``mean-prediction`` : estimates k predictions per sample, based on
+                each of the k-fold cross-validation classifiers, and average
+                these predictions into a single estimate per sample.
```

Comment:
I think we just trimmed down the prediction API with @kingjr -- let's make sure to be consistent.